### PR TITLE
V1 add marker proposal

### PR DESCRIPTION
**Pull request by llama-del-rey**
## Description

closes: #1287

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,7 +37,11 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 ## Unreleased
 
-* nothing
+### Improvements:
+* Migrate `AddMarkerProposal` to v1 message based gov proposal [PR 1321](https://github.com/provenance-io/provenance/pull/1321)
+
+### Client Breaking
+* Migrate `AddMarkerProposal` to v1 message based gov proposal [PR 1321](https://github.com/provenance-io/provenance/pull/1321)
```
Comment (SpicyLemon) [inline CHANGELOG.md:44]: When fixing the conflicts here, make sure to add an empty line after the heading and before the entries.

File: CHANGELOG.md [hunk 2]
```diff
@@ -48,6 +52,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * Bump tendermint to Notional's v0.34.25 (from base repo's v0.34.24) [PR 1348](https://github.com/provenance-io/provenance/pull/1348).
 * Bump Cosmos-SDK to v0.46.8-pio-3 (from [v0.46.7-pio-2](https://github.com/provenance-io/cosmos-sdk/compare/v0.46.7-pio-2...v0.46.8-pio-3)) [PR 1348](https://github.com/provenance-io/provenance/pull/1348).
 * Set the immediate sanction/unsanction min deposits to 1,000,000 hash in the `paua` upgrade [PR 1345](https://github.com/provenance-io/provenance/pull/1345).
+* Migrate `AddMarkerProposal` to v1 message based gov proposal [PR 1321](https://github.com/provenance-io/provenance/pull/1321)
```
Comment (SpicyLemon) [inline CHANGELOG.md:55]: And this shouldn't be down here.

File: x/marker/module.go [hunk 3]
```diff
@@ -91,16 +92,19 @@ type AppModule struct {
 	accountKeeper  authkeeper.AccountKeeper
 	bankKeeper     bankkeeper.Keeper
 	feegrantKeeper feegrantkeeper.Keeper
+	govKeeper      govkeeper.Keeper
```
Comment (SpicyLemon) [inline x/marker/module.go:95]: I think you want `types.GovKeeper` here (and as an argument to `NewAppModule`. That way, this module (or at least this file) doesn't have a direct dependency on the gov module.

File: x/marker/simulation/operations.go [hunk 4]
```diff
@@ -49,13 +55,23 @@ SetDenomMetadata
 
 // WeightedOperations returns all the operations from the module with their respective weights
 func WeightedOperations(
-	appParams simtypes.AppParams, cdc codec.JSONCodec, k keeper.Keeper, ak authkeeper.AccountKeeperI, bk bankkeeper.Keeper,
+	appParams simtypes.AppParams, cdc codec.JSONCodec, protoCodec *codec.ProtoCodec, k keeper.Keeper, ak authkeeper.AccountKeeper, bk bankkeeper.Keeper, gk govkeeper.Keeper,
```
Comment (SpicyLemon) [inline x/marker/simulation/operations.go:58]: Here too with the `types.GovKeeper` interface.

**Review by nullpointer0x00 — CHANGES_REQUESTED** [review]
We should add a cli to submit a gov proposal tx.

**Review by nullpointer0x00 — DISMISSED** [review]

**Review by nullpointer0x00 — APPROVED** [review]

File: proto/provenance/marker/v1/tx.proto [hunk 5]
```diff
@@ -203,7 +205,7 @@ message MsgSetDenomMetadataResponse {}
 // MsgAddFinalizeActivateMarkerRequest defines the Msg/AddFinalizeActivateMarker request type
 message MsgAddFinalizeActivateMarkerRequest {
   cosmos.base.v1beta1.Coin amount = 1
-      [(gogoproto.nullable) = false, (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Coin"];
+  [(gogoproto.nullable) = false, (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Coin"];
```
Comment (SpicyLemon) [inline proto/provenance/marker/v1/tx.proto:208]: IntelliJ annoyingly removes those spaces, but they should be there.

**Review by SpicyLemon — APPROVED** [review]
Approving even though there's conflicts that'll dismiss it. But it looks good right now.

**Comment by SpicyLemon** [comment]
I've copied this branch into one in this repo, and fixed it up a bit: [dwedul/add-marker-proposal](https://github.com/provenance-io/provenance/compare/dwedul/add-marker-proposal). It's not yet ready for a PR, but I'm going to close this PR now anyway.

**Pull request closed without merging**